src/runtime.cc

```diff
@@ -102,13 +102,19 @@ namespace detail {
 		MPI_Comm_size(MPI_COMM_WORLD, &world_size);
 		num_nodes = world_size;
 
+		cfg = std::make_unique<config>(argc, argv);
+		if(is_dry_run()) {
```

Comment:
```suggestion
		if(cfg->is_dry_run()) {
```

A minor thing, but otherwise it's not clear that config must exist before `runtime::is_dry_run` can be called.